File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.cpp
class Solution {
public:
    vector<int> findMissingAndRepeatedValues(vector<vector<int>>& v) {
        int n = v.size();
        vector<int> mp(n * n + 1, 0);
        
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < v[i].size(); j++) { 
                mp[v[i][j]]++;
            }
        }

        vector<int> ans(2, 0);

        for (int i = 1; i <= n * n; i++) {  
            if (mp[i] == 2) {
                ans[0] = i; 
            } else if (mp[i] == 0) {
                ans[1] = i; 
            }
        }

        return ans;
    }
};

File: 3227-find-missing-and-repeated-values/3227-find-missing-and-repeated-values.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    let v = grid.flat(); 
    let n = v.length;
    let freq = new Map();
    let duplicate, missing;

    for (let num of v) {
        freq.set(num, (freq.get(num) || 0) + 1);
    }

    for (let i = 1; i <= n; i++) {
        if (freq.get(i) === 2) {
            duplicate = i;
        }
        if (!freq.has(i)) {
            missing = i;
        }
    }

    return [duplicate, missing];
};
